### ISSUE TITLE
`Transformers` : Error while fitting TFBertForSequenceClassification model

### ISSUE THREAD

**Issue by ImRobot777** [microsoft/tensorflow-directml-plugin#251]
Hi,

First of all thanks a lot making directML compatible with TensorFlow > 2 !
As the Transformers lib needs TensorFlow >= 2.3 i had got hope my TFBertForSequenceClassification model was working with my AMD GPU card (Rx 6800)

Unfortuntly, this is not the case. (important detail : it takes a crazy long time but works on CPU with standard Tensorflow lib)

My env: 
windows 11 PRO 64bit : 21H2
python 3.8.13
tensorflow-cpu 2.9.1
tensorflow-directml-plugin 0.0.1.dev220621
CPU : Ryzen 5600X, 
GPU : Rx 6800

The GPU is rightly recognized: 
```python
tf.config.list_physical_devices('GPU')
[PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
```

My code : 
```python
bert_model = TFBertForSequenceClassification.from_pretrained('bert-base-uncased', num_labels=2)

callbacks = [tf.keras.callbacks.ModelCheckpoint(filepath=model_save_path,
                                                save_weights_only=True,
                                                monitor='val_loss',
                                                mode='min',
                                                save_best_only=True),
             keras.callbacks.TensorBoard(log_dir=log_dir)]

print('\nBert Model', bert_model.summary())

loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
metric = [tf.keras.metrics.SparseCategoricalAccuracy('accuracy')]
optimizer = tf.keras.optimizers.Adam(learning_rate=2e-5,epsilon=1e-08)

bert_model.compile(loss=loss, optimizer=optimizer, metrics=metric)

#OUTPUT : 
Model: "tf_bert_for_sequence_classification_3"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 bert (TFBertMainLayer)      multiple                  109482240 
                                                                 
 dropout_151 (Dropout)       multiple                  0         
                                                                 
 classifier (Dense)          multiple                  1538      
                                                                 
=================================================================
Total params: 109,483,778
Trainable params: 109,483,778
Non-trainable params: 0
_________________________________________________________________

```

Above code works well but when reaching the fit : 

```python
history=bert_model.fit([X_train, Mask_Train], 
                       y_train,
                       batch_size=32,
                       epochs=EPOCHS,
                       validation_data=([X_test, Mask_test], y_test),
                       callbacks=callbacks)
```

`The GPU VRAM begins to receive data` (i'm monitoring it via Radeon Adrenalin Software) and suddenly an error message (see below) appears !

The Error Message : 

```python
Epoch 1/3
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
File <timed exec>:2, in <module>

File ~\anaconda3\envs\P7_Bert_TF29_PYT38\lib\site-packages\keras\utils\traceback_utils.py:67,
 in filter_traceback.<locals>.error_handler(*args, **kwargs)
     65 except Exception as e:  # pylint: disable=broad-except
     66   filtered_tb = _process_traceback_frames(e.__traceback__)
---> 67   raise e.with_traceback(filtered_tb) from None
     68 finally:
     69   del filtered_tb

File ~\anaconda3\envs\P7_Bert_TF29_PYT38\lib\site-packages\tensorflow\python\eager\execute.py:54, 
 in quick_execute(op_name, num_outputs, inputs, attrs, ctx, name)
     52 try:
     53   ctx.ensure_initialized()
---> 54   tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
     55                                       inputs, attrs, num_outputs)
     56 except core._NotOkStatusException as e:
     57   if name is not None:

InvalidArgumentError: Cannot assign a device for operation tf_bert_for_sequence_classification_3/bert/embeddings/Gather: 
Could not satisfy explicit device specification '' 
because the node {{colocation_node tf_bert_for_sequence_classification_3/bert/embeddings/Gather}} was colocated 
with a group of nodes that required incompatible device '/job:localhost/replica:0/task:0/device:GPU:0'. 
All available devices [/job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:GPU:0]. 
Colocation Debug Info:
Colocation group had the following types and supported devices: 
Root Member(assigned_device_name_index_=2 requested_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' 
assigned_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' 
resource_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' supported_device_types_=[CPU] possible_devices_=[]

StridedSlice: CPU 
Unique: GPU CPU 
Shape: GPU CPU 
_Arg: GPU CPU 
ResourceGather: GPU CPU 
Const: GPU CPU 
UnsortedSegmentSum: CPU 
Mul: GPU CPU 
ReadVariableOp: GPU CPU 
AssignVariableOp: GPU CPU 
ResourceScatterAdd: GPU CPU 
Sqrt: GPU CPU 
AddV2: GPU CPU 
RealDiv: GPU CPU 
AssignSubVariableOp: GPU CPU 
NoOp: GPU CPU 

Colocation members, user-requested devices, and framework assigned devices, if any:
  tf_bert_for_sequence_classification_3_bert_embeddings_gather_resource (_Arg)  
       framework assigned device=/job:localhost/replica:0/task:0/device:GPU:0
  adam_adam_update_readvariableop_resource (_Arg)  
       framework assigned device=/job:localhost/replica:0/task:0/device:GPU:0
  adam_adam_update_readvariableop_2_resource (_Arg)  
       framework assigned device=/job:localhost/replica:0/task:0/device:GPU:0
  tf_bert_for_sequence_classification_3/bert/embeddings/Gather (ResourceGather) 
  Adam/Adam/update/Unique (Unique) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/Shape (Shape) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/strided_slice/stack (Const) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/strided_slice/stack_1 (Const) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/strided_slice/stack_2 (Const) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/strided_slice (StridedSlice) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/UnsortedSegmentSum (UnsortedSegmentSum) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/mul (Mul) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/ReadVariableOp (ReadVariableOp) 
  Adam/Adam/update/mul_1 (Mul) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/AssignVariableOp (AssignVariableOp) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/ResourceScatterAdd (ResourceScatterAdd) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/ReadVariableOp_1 (ReadVariableOp) 
  Adam/Adam/update/mul_2 (Mul) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/mul_3 (Mul) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/ReadVariableOp_2 (ReadVariableOp) 
  Adam/Adam/update/mul_4 (Mul) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/AssignVariableOp_1 (AssignVariableOp) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/ResourceScatterAdd_1 (ResourceScatterAdd) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/ReadVariableOp_3 (ReadVariableOp) 
  Adam/Adam/update/Sqrt (Sqrt) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/mul_5 (Mul) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/add (AddV2) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/truediv (RealDiv) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/AssignSubVariableOp (AssignSubVariableOp) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/group_deps/NoOp (NoOp) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/group_deps/NoOp_1 (NoOp) /job:localhost/replica:0/task:0/device:GPU:0
  Adam/Adam/update/group_deps (NoOp) /job:localhost/replica:0/task:0/device:GPU:0

      [[{{node tf_bert_for_sequence_classification_3/bert/embeddings/Gather}}]] 
      [Op:__inference_train_function_57566]
```

Thanks in advance for any Help !
Have a good day.

**Comment by PatriceVignola** [comment]
Hey @ImRobot777, I can't be 100% sure without having a full repro, but from the log above it looks like all that's missing is the int32 data type support for `StridedSlice` and `UnsortedSegmentSum`. We already have implementations for those ops, so all that's missing should be to add support for int32. We are currently planning to release a new version of the plugin shortly after the upcoming TensorFlow 2.10 release, so you shouldn't have to wait too long.

If I provide you with a development wheel after the change has been merged in, would you be willing to try it out and let me know if it fixes your issue? Or if you're comfortable with building from source that should also work.

**Comment by ImRobot777** [comment]
Hi,
Thanks a lot for your help @PatriceVignola 

Of course i wanna try with a development wheel, i'll be so relieved and happy that I finally got my code to work !
You just cannot imagine how many different test i've done (all possible combinations with Tensorflow 1.15 with tensorflow-directml to TS 2.9(all possible variant std, -cpu, -gpu)  with tensorflow-directml-plugin and different compatible versions of python... tried to se soft_placement=true in the head of my jupyter notebook too...), the code and env from this issue form are the "best" results i've had.

Unfortunatly, the last time i've compiled sources was 20 years before for C++ student project :-P
That means that i do not know how to build package from source but if you have in mind a simple tutorial to build from source, i'll do it !

If it fix my issue, you'll be the first to know it :-)
Have a good day.
Guillaume

**Comment by PatriceVignola** [comment]
We have a the first nightly build that should contain the fixes: https://github.com/microsoft/tensorflow-directml-plugin/actions/runs/2887210312

Please try it out and let me know how it works for you! Note that we haven't been focusing on Transformer models performance thus far so we don't know how well we perform on them, but this is something we'll be focusing on in the next few months. Our main concern right now is making sure that they run to completion without crashing :)

**Comment by ImRobot777** [comment]
Hi Patrice,
I'm so sorry haven't seen your answer this morning.
I was so late on my project that i've had to replace my AMD GPU (that i love so much) by a Nvidia one, 2 times less powerfull but run well my code at least...

I do not know when i'll have the time to do another GPU Switch (+drivers, cuda and so on) but when i'll do it, i'll test the built immediatly even if i'm not really sure what i'm supposed to do with the link in your previous response.

It looks like a docker file and i've never tried docker yet, will it work on windows 11 ?
Best Regards, have a good day.
Guillaume

**Comment by PatriceVignola** [comment]
No problem! tensorflow-directml-plugin is also compatible with Nvidia. The link I posted is just a page where you can download the package wheels. For example, if you want to test the Windows package for python 3.10, you would download [this zip file](https://github.com/microsoft/tensorflow-directml-plugin/suites/7884116315/artifacts/334784432), unzip it and call `pip install tensorflow_directml_plugin-0.0.1-cp310-cp310-win_amd64.whl`.

**Comment by ImRobot777** [comment]
My bad, thank you for the explanations Patrice !

I've got just 2 points to add, a note and a question:

1/ I've only installed package nammed tensorflow and note tensorflow-gpu, cause since tensorflow 2.5, the standart package manage both CPU and GPU and it works well with the RTX 3060

2/ I've forgotten that but as you mention it, i remember now that directml-plugin can work with nvidia GPU.
Is that mean that without switching again my GPUs, i could reproduce the issue on the Nvidia card with the current directML-plugin and then see if it work with the nightly version of your link ???

Best Regards,
Guillaume

**Comment by PatriceVignola** [comment]
Yes! You don't need to switch GPUs to try it out. The only difference is that you'll need the `tensorflow-cpu` package instead of `tensorflow` alongside the plugin and you don't have to install or configure CUDA or cuDNN.

**Comment by ImRobot777** [comment]
Great !
But,  indeed, i needed installing CUDA and cuDNN for the first try of my new Nvidia GPU, so they are already installed on my OS ?
So, will i still be able to reproduce the issue ??? (i mean in a new conda virtual env of course with tensorflow-directml-plugin)

**Comment by PatriceVignola** [comment]
You will be able to reproduce the issue. What I meant is that you don't need CUDA or cuDNN to make `tensorflow-directml-plugin` work with your Nvidia card :)

**Comment by ImRobot777** [comment]
OK, thanks a lot,
I'll try to test it as soon as possible !
And come back to you

**Comment by ImRobot777** [comment]
Well, 
I've test it and the installation worked well !
But unfortunatly lot of fatal errors appeared when importing the package : 

one of them
```Python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [4], in <cell line: 28>()
     24 import tensorflow as tf
     25 #tf.config.set_soft_device_placement(True)
     26 #tf.compat.v1.config.set_soft_device_placement(True)
---> 28 import tensorflow_hub as hub
     30 import tensorflow.keras as keras
     32 from keras.callbacks import ModelCheckpoint

File ~\anaconda3\envs\P7_DirectML_38\lib\site-packages\tensorflow_hub\__init__.py:85, in <module>
     74   if (distutils.version.LooseVersion(tf.__version__) <
     75       distutils.version.LooseVersion(required_tensorflow_version)):
     76     raise ImportError(
     77         "\n\nThis version of tensorflow_hub requires tensorflow "
     78         "version >= {required}; Detected an installation of version {present}. "
   (...)
     82             required=required_tensorflow_version,
     83             present=tf.__version__))
---> 85 _ensure_tf_install()
     88 from tensorflow_hub.estimator import LatestModuleExporter
     89 from tensorflow_hub.estimator import register_module_for_export

File ~\anaconda3\envs\P7_DirectML_38\lib\site-packages\tensorflow_hub\__init__.py:74, in _ensure_tf_install()
     63 #
     64 # Update this whenever we need to depend on a newer TensorFlow release.
     65 #
   (...)
     71 # and character class transitions.
     72 #
     73 required_tensorflow_version = "1.15.0"
---> 74 if (distutils.version.LooseVersion(tf.__version__) <
     75     distutils.version.LooseVersion(required_tensorflow_version)):
     76   raise ImportError(
     77       "\n\nThis version of tensorflow_hub requires tensorflow "
     78       "version >= {required}; Detected an installation of version {present}. "
   (...)
     82           required=required_tensorflow_version,
     83           present=tf.__version__))

AttributeError: module 'tensorflow' has no attribute '__version__'
```

Then another try gives me a another error that said There is no `Keras` attribute in tensorflow module.

And finally, another one that shows incompatibility with `transformers` package

i do not manage to reproduce the 2 last errors after having re-installed the virtual env, but they could appear again if your team fix the first issue.

Wait and see, hope it will be fixed soon.
Have a goo WE

**Comment by PatriceVignola** [comment]
It looks like some of your installed packages were conflicting with each other (e.g. `required_tensorflow_version  = "1.15.0"` in tensorflow_hub). I'm not sure how this package made it into your virtual environment but `tensorflow-directml-plugin` only depends on `tensorflow-cpu>=2.10`, so it couldn't have been the cause.

**Comment by ImRobot777** [comment]
See the content of my environment and search for tensorflow, you'll see that 2.10 has been installed : 

```yaml
name: P7_DirectML_38
channels:
  - defaults
dependencies:
  - argon2-cffi=21.3.0=pyhd3eb1b0_0
  - argon2-cffi-bindings=21.2.0=py38h2bbff1b_0
  - asttokens=2.0.5=pyhd3eb1b0_0
  - attrs=21.4.0=pyhd3eb1b0_0
  - backcall=0.2.0=pyhd3eb1b0_0
  - beautifulsoup4=4.11.1=py38haa95532_0
  - bleach=4.1.0=pyhd3eb1b0_0
  - ca-certificates=2022.07.19=haa95532_0
  - certifi=2022.6.15=py38haa95532_0
  - cffi=1.15.1=py38h2bbff1b_0
  - colorama=0.4.5=py38haa95532_0
  - console_shortcut=0.1.1=4
  - debugpy=1.5.1=py38hd77b12b_0
  - decorator=5.1.1=pyhd3eb1b0_0
  - defusedxml=0.7.1=pyhd3eb1b0_0
  - entrypoints=0.4=py38haa95532_0
  - executing=0.8.3=pyhd3eb1b0_0
  - importlib_resources=5.2.0=pyhd3eb1b0_1
  - ipykernel=6.9.1=py38haa95532_0
  - ipython=8.4.0=py38haa95532_0
  - ipython_genutils=0.2.0=pyhd3eb1b0_1
  - jedi=0.18.1=py38haa95532_1
  - jinja2=3.0.3=pyhd3eb1b0_0
  - jsonschema=4.4.0=py38haa95532_0
  - jupyter_client=7.2.2=py38haa95532_0
  - jupyter_core=4.10.0=py38haa95532_0
  - jupyterlab_pygments=0.1.2=py_0
  - libsodium=1.0.18=h62dcd97_0
  - markupsafe=2.1.1=py38h2bbff1b_0
  - matplotlib-inline=0.1.2=pyhd3eb1b0_2
  - mistune=0.8.4=py38he774522_1000
  - nbclient=0.5.13=py38haa95532_0
  - nbconvert=6.4.4=py38haa95532_0
  - nbformat=5.3.0=py38haa95532_0
  - nest-asyncio=1.5.5=py38haa95532_0
  - notebook=6.4.12=py38haa95532_0
  - openssl=1.1.1q=h2bbff1b_0
  - packaging=21.3=pyhd3eb1b0_0
  - pandocfilters=1.5.0=pyhd3eb1b0_0
  - parso=0.8.3=pyhd3eb1b0_0
  - pickleshare=0.7.5=pyhd3eb1b0_1003
  - prometheus_client=0.14.1=py38haa95532_0
  - prompt-toolkit=3.0.20=pyhd3eb1b0_0
  - pure_eval=0.2.2=pyhd3eb1b0_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pygments=2.11.2=pyhd3eb1b0_0
  - pyparsing=3.0.4=pyhd3eb1b0_0
  - pyrsistent=0.18.0=py38h196d8e1_0
  - python=3.8.13=h6244533_0
  - python-dateutil=2.8.2=pyhd3eb1b0_0
  - python-fastjsonschema=2.15.1=pyhd3eb1b0_0
  - pywin32=302=py38h2bbff1b_2
  - pywinpty=2.0.2=py38h5da7b33_0
  - pyzmq=23.2.0=py38hd77b12b_0
  - send2trash=1.8.0=pyhd3eb1b0_1
  - setuptools=61.2.0=py38haa95532_0
  - six=1.16.0=pyhd3eb1b0_1
  - soupsieve=2.3.1=pyhd3eb1b0_0
  - sqlite=3.39.2=h2bbff1b_0
  - stack_data=0.2.0=pyhd3eb1b0_0
  - terminado=0.13.1=py38haa95532_0
  - testpath=0.6.0=py38haa95532_0
  - tornado=6.1=py38h2bbff1b_0
  - traitlets=5.1.1=pyhd3eb1b0_0
  - typing-extensions=4.3.0=py38haa95532_0
  - typing_extensions=4.3.0=py38haa95532_0
  - vc=14.2=h21ff451_1
  - vs2015_runtime=14.27.29016=h5e58377_2
  - wcwidth=0.2.5=pyhd3eb1b0_0
  - webencodings=0.5.1=py38_1
  - wheel=0.37.1=pyhd3eb1b0_0
  - wincertstore=0.2=py38haa95532_2
  - winpty=0.4.3=4
  - zeromq=4.3.4=hd77b12b_0
  - zipp=3.8.0=py38haa95532_0
  - pip:
    - absl-py==1.2.0
    - astunparse==1.6.3
    - cachetools==5.2.0
    - charset-normalizer==2.1.1
    - cycler==0.11.0
    - filelock==3.8.0
    - flatbuffers==2.0
    - fonttools==4.36.0
    - gast==0.4.0
    - google-auth==2.11.0
    - google-auth-oauthlib==0.4.6
    - google-pasta==0.2.0
    - grpcio==1.47.0
    - h5py==3.7.0
    - huggingface-hub==0.8.1
    - idna==3.3
    - importlib-metadata==4.12.0
    - joblib==1.1.0
    - keras==2.10.0rc1
    - keras-preprocessing==1.1.2
    - kiwisolver==1.4.4
    - libclang==14.0.6
    - markdown==3.4.1
    - matplotlib==3.5.3
    - numpy==1.23.2
    - oauthlib==3.2.0
    - opt-einsum==3.3.0
    - pandas==1.4.3
    - pillow==9.2.0
    - pip==22.2.2
    - protobuf==3.19.4
    - pyasn1==0.4.8
    - pyasn1-modules==0.2.8
    - pytz==2022.2.1
    - pyyaml==6.0
    - regex==2022.8.17
    - requests==2.28.1
    - requests-oauthlib==1.3.1
    - rsa==4.9
    - scikit-learn==1.1.2
    - scipy==1.9.0
    - sklearn==0.0
    - tensorboard==2.9.1
    - tensorboard-data-server==0.6.1
    - tensorboard-plugin-wit==1.8.1
    - tensorflow-cpu==2.10.0rc1
    - tensorflow-directml-plugin==0.0.1
    - tensorflow-estimator==2.10.0rc0
    - tensorflow-hub==0.12.0
    - tensorflow-intel==2.10.0rc1
    - tensorflow-io-gcs-filesystem==0.26.0
    - termcolor==1.1.0
    - threadpoolctl==3.1.0
    - tokenizers==0.12.1
    - tqdm==4.64.0
    - transformers==4.21.1
    - urllib3==1.26.11
    - werkzeug==2.2.2
    - wrapt==1.14.1

```

**Comment by ImRobot777** [comment]
May be, i should not run `python -m pip install --upgrade pip`  first ???

**Comment by ImRobot777** [comment]
PS : I just forgot to mention that with current stable tensorflow-directml-plugin, i've managed to reproduce with the Nvidia card the exact same error that with my AMD card

**Comment by ImRobot777** [comment]
Well,
By uninstalling tensorflow-hub package and removing it from my packages import, i manage to get the 1st error message i saw with this directml nightly package : 

```python
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Input In [2], in <cell line: 28>()
     24 import tensorflow as tf
     25 #tf.config.set_soft_device_placement(True)
     26 #tf.compat.v1.config.set_soft_device_placement(True)
---> 28 import tensorflow_hub as hub
     30 import tensorflow.keras as keras
     32 from keras.callbacks import ModelCheckpoint

ModuleNotFoundError: No module named 'tensorflow_hub'
```

**Comment by PatriceVignola** [comment]
I think the version check that tensorflow-hub does for the version of TensorFlow unfortunately just doesn't work with `rc1` versions or development versions, which explains the error message above.

**Comment by ImRobot777** [comment]
> 

Yeah, i thought about something like that, that's why i've uninstalled it and got the first error message (about Keras) i talked about before, see above ;-)

**Comment by ImRobot777** [comment]
> I think the version check that tensorflow-hub does for the version of TensorFlow unfortunately just doesn't work with `rc1` versions or development versions, which explains the error message above.

Yeah, i thought about something like that, that's why i've uninstalled it and got the first error message (about Keras) i talked about before, see above ;-)

**Comment by ImRobot777** [comment]
> > I think the version check that tensorflow-hub does for the version of TensorFlow unfortunately just doesn't work with `rc1` versions or development versions, which explains the error message above.
> 
> Yeah, i thought about something like that, that's why i've uninstalled it and got the first error message (about Keras) i talked about before, see above ;-)

Ho sorry, as i was coming for news, by reading my above answear, i saw that it has probably posted the wrong screen shot, or there is another issue as i've removed tensorflow_hub package.

I must go deeper in my tests and analysis, but i'm not having access to my working PC right now.
I'll came back with news as soon as possible !

**Comment by ImRobot777** [comment]
Hi, i'm back for tests.
Do i have to get a nightly package of directML-plugin again or the stable published one ?

**Comment by ImRobot777** [comment]
So with the new nightly, i got this error : 

'''python
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Input In [3], in <cell line: 30>()
     24 import tensorflow as tf
     25 #tf.config.set_soft_device_placement(True)
     26 #tf.compat.v1.config.set_soft_device_placement(True)
     27 
     28 #import tensorflow_hub as hub
---> 30 import tensorflow.keras as keras
     32 from keras.callbacks import ModelCheckpoint
     34 #BERT

ModuleNotFoundError: No module named 'tensorflow.keras'
'''

**Comment by PatriceVignola** [comment]
`tensorflow-directml-plugin` has nothing to do with keras, so I think there's some problems in your environment. At this point, I think it's better to wait for the release of tensorflow-cpu 2.10.0 (which should release any day now) and create a clean environment when it's out.

**Comment by ImRobot777** [comment]
I agree but my virtual env is completly new and the command lines i've used to install the different package of my project are the same than the package of my stable env that make tensorflow(standart not directML) works with my RTX 3060.
The only difference between these 2 virtual env is the tensorflow_directml_plugin-0.0.1-cp38-cp38-win_amd64.whl that is replacing the standar tensorflow package.

For reminding, when i use the official directML-plugin (so not the nightly) i can import my package without any problems.
The only issye is the first one for which i've created this ticket.

**Comment by ImRobot777** [comment]
Hi,

Well, so, i've just tested with tensorflow-cpu and even before reaching the .fit(...) code i've got a Jupyter kernel crash, see below

![image](https://user-images.githubusercontent.com/92150826/189932811-78f3c381-1166-46a1-8066-fa9d15c75269.png)

It seems that the crash is occuring when trying to execute this line: 
```python
tf.keras.metrics.SparseCategoricalAccuracy('accuracy')
```

**Comment by PatriceVignola** [comment]
@ImRobot777 we just released [version 0.1.0.dev220928](https://pypi.org/project/tensorflow-directml-plugin/0.1.0.dev220928/) which adds the missing `UnsortedSegmentSum` and `StridedSlice` operators. From our internal transformers testing, we were able to run some of the example transformer models. If you're still having problems after installing this package and `tensorflow-cpu==2.10.0` in a brand new environment, please let us know.

**Comment by ImRobot777** [comment]
Hi,

Well, so, i've just tested with tensorflow-cpu 2.10 and the last directML pluging and this time the .fit(...) has been reached and even started, but unfortunatly stopped by a Jupyter kernel crash (again...) , see below

![image](https://user-images.githubusercontent.com/92150826/200330138-db39b8a9-b046-4b9c-bd05-7d77fa3b4728.png)


Below last command lines executed before the crash: 
![image](https://user-images.githubusercontent.com/92150826/200330771-e196edfd-4729-4259-ac28-1a2920a3d2dd.png)

Hope you'll find the reason of that crash.
Have a good day,

Best Regards
Guillaume
